test/e2e/storage/volume_metrics.go

```diff
@@ -673,7 +673,7 @@ func waitForDetachAndGrabMetrics(ctx context.Context, oldMetrics *storageControl
 		oldDetachCount = 0
 	}
 
-	verifyMetricFunc := func() (bool, error) {
+	verifyMetricFunc := func(ctx context.Context) (bool, error) {
 		updatedMetrics, err := metricsGrabber.GrabFromControllerManager(ctx)
```

Comment:
what ctx uses it with this change, the one from passed as parameter in the func, right?

Reply:
I don't think that is changing anything, is just curiosity

Reply:
Correct.  I think from a style perspective we should enforce that closures use the context provided in the closure signature in preference to the closure of the enclosing scope.  That seems consistent with other usage, and given that not all conditions passed to a wait function will be closures, prevents reviewers from being surprised.  Also, if you are closing over two contexts, and you specifically try to use the parent scope's context to avoid cancellation... you are probably doing something really complicated and scary, and should have a nice chunky comment explaining why.